packages/test/test-version-utils/src/compatUtils.ts

```diff
@@ -42,6 +49,79 @@ import {
  */
 export const TestDataObjectType = "@fluid-example/test-dataStore";
 
+/**
+ * This function modifies container runtime options according to a version of runtime used.
+ * If a version of runtime does not support some options, they are removed.
+ * If a version runtime supports some options, such options are enabled to increase a chance of
+ * hitting feature set controlled by such options, and thus increase chances of finding product bugs.
+ *
+ * @param version - a version of container runtime to be used in test
+ * @param optionsArg - input runtime options (optional)
+ * @returns - runtime options that should be used with a given version of container runtime
+ * @internal
+ */
+function filterRuntimeOptionsForVersion(
+	version: string,
+	optionsArg: IContainerRuntimeOptions = {
+		summaryOptions: {
+			summaryConfigOverrides: {
+				...DefaultSummaryConfiguration,
+				...{
+					initialSummarizerDelayMs: 0,
+				},
+			},
+		},
+	},
+) {
+	let options = { ...optionsArg };
+
+	// No test fails with this option, it allows us to validate properly expectations and
+	// implementation of services
+	options.loadSequenceNumberVerification = "close";
+
+	const compressorDisabled: ICompressionRuntimeOptions = {
+		minimumBatchSizeInBytes: Number.POSITIVE_INFINITY,
+		compressionAlgorithm: CompressionAlgorithms.lz4,
+	};
+
+	// These is the "maximum" config.
+	const {
+		compressionOptions = {
+			minimumBatchSizeInBytes: 200,
+			compressionAlgorithm: CompressionAlgorithms.lz4,
+		},
+		chunkSizeInBytes = 200,
+		enableRuntimeIdCompressor = "on",
+		enableGroupedBatching = true,
+	} = options;
+
+	if (version === "1.3.7") {
+		options.compressionOptions = undefined;
+		options.enableGroupedBatching = false;
+		options.enableRuntimeIdCompressor = "off";
+		options.maxBatchSizeInBytes = undefined;
+		options.chunkSizeInBytes = Number.POSITIVE_INFINITY; // disabled
+	} else if (version.includes("2.0.0-rc.1.0.4")) {
+		options = {
+			...options,
+			compressionOptions: compressorDisabled, // Can't use compression, need https://github.com/microsoft/FluidFramework/pull/20111 fix
+			chunkSizeInBytes: Number.POSITIVE_INFINITY, // disabled, need https://github.com/microsoft/FluidFramework/pull/20115 fix
+			enableRuntimeIdCompressor,
+			enableGroupedBatching,
+		};
+	} else if (version.includes("2.0.0-rc.2.")) {
+		options = {
+			...options,
+			compressionOptions,
+			chunkSizeInBytes,
+			enableRuntimeIdCompressor,
+			enableGroupedBatching,
+		};
```

Comment:
FYI - these configs need to be moved out somewhere else, and ideally be coming from versioned runtime packages  in the future. I'll come back to it after I'm done with somewhat related https://github.com/microsoft/FluidFramework/pull/20174 and next steps of exposing all these configs for AzureClient